### PR TITLE
Bp 37: Add admin login

### PR DESCRIPTION
**Pull request by MrSachin7**

### CODE REVIEW

File: src/Data/converters/UserConverter.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using Data.Entities;
+using Domain.utils;
+using Dto;
+using Dto.tools;
+
+namespace Data.converters;
+
+public static class UserConverter {
+    public static UserDto? ToEntity(AdminEntity? adminEntity) {
```
Comment (himal28924) [inline src/Data/converters/UserConverter.cs]: `ToDto`

File: src/Domain/auth/AuthenticationDomain.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    public async Task<UserDto> ValidateAdmin(LoginRequestDto loginRequest) {
+        CheckForValidEmail(loginRequest.Email);
+        CheckForValidPassword(loginRequest.Password);
+
+        UserDto? userFromDatabase = await _authDao.GetUserByEmail(loginRequest.Email);
+        if (userFromDatabase is null) {
+            throw new ExceptionWithErrorCode(ErrorCode.NotFound, ErrorMessages.EmailDoesntExist);
+        }
+
+        bool doesPasswordMatch = VerifyPassword(loginRequest.Password, userFromDatabase.Password);
```
Comment (himal28924) [inline src/Domain/auth/AuthenticationDomain.cs:25]: `        if (VerifyPassword(loginRequest.Password, userFromDatabase.Password))
        {
            return userFromDatabase;
        }
        else
        {
            throw new ExceptionWithErrorCode(ErrorCode.BadRequest, ErrorMessages.IncorrectPassword);
        }`

File: src/Data/converters/UserConverter.cs (hunk 1 above)
Comment (MrSachin7) [reply src/Data/converters/UserConverter.cs]: Ah, had to cancel the prev commit, so this remained here. Would fix .✌️

File: src/Domain/auth/AuthenticationDomain.cs (hunk 2 above)
Comment (MrSachin7) [reply src/Domain/auth/AuthenticationDomain.cs:25]: I disagree. The current way is more clear.

**Review by himal28924 — APPROVED** [review]
NICE 👍

**Pull request merged**